deepspeed/pt/deepspeed_light.py

```diff
@@ -979,7 +979,12 @@ def allreduce_no_retain(self, bucket, numel_per_bucket=500000000):
     def buffered_allreduce_fallback(self, grads=None, elements_per_buffer=500000000):
         grads = []
         for param_name, param in self.module.named_parameters():
-            if param.grad is not None:
+            if param.grad is None:
```

Comment:
May I suggest to add a comment here to explain why this change was necessary? It may be useful for future readers of the code who may not have had the experience we ran into. Thanks!